pycheribuild/projects/project.py

```diff
@@ -1770,24 +1780,22 @@ def update(self, current_project: "Project", *, src_dir: Path, base_project_sour
             return
 
         # Handle forced branches now that we have fetched the latest changes
-        if src_dir.exists() and self.force_branch:
-            default_branch = self.get_default_branch(current_project, include_per_target=True)
-            assert default_branch, "default_branch must be set if force_branch is true!"
-            # TODO: move this to Project so it can also be used for other targets
-            status = run_command("git", "status", "-b", "-s", "--porcelain", "-u", "no",
-                                 capture_output=True, print_verbose_only=True, cwd=src_dir,
-                                 run_in_pretend_mode=_PRETEND_RUN_GIT_COMMANDS)
-            if status.stdout.startswith(b"## ") and not status.stdout.startswith(
-                    b"## " + default_branch.encode("utf-8") + b"..."):
-                current_branch = status.stdout[3:status.stdout.find(b"...")].strip()
-                current_project.warning("You are trying to build the", current_branch.decode("utf-8"),
+        if src_dir.exists() and (self.force_branch or self.old_branches):
+            if self.force_branch:
+                assert self.old_branches is None, "cannot set both force_branch and old_branches"
+                default_branch = self.get_default_branch(current_project, include_per_target=True)
+                assert default_branch, "default_branch must be set if force_branch is true!"
+            current_branch = self.get_current_branch(src_dir).decode("utf-8")
+            if self.old_branches:
+                default_branch = self.old_branches.get(current_branch)
```

Comment:
This won't work if current_branch isn't in old_branches; it won't raise an exception like `[]`, instead it'll give None and blow up later. Easiest solution is to add current_branch as the optional default value argument.

Reply:
Oh, never mind, you already check for None... though if you want to have `"foo": None` mean `self._default_branch` then you do need to be able to distinguish the cases

Reply:
I check for None below (I purposely didn't want an exception)

Reply:
Yeah, this is why using None to mean default branch is a bit messier.